homework/src/index.js

```diff
@@ -33,14 +33,123 @@
   function main(url) {
     fetchJSON(url, (err, data) => {
       const root = document.getElementById('root');
+      const header = createAndAppend('header', root);
+      createAndAppend('p', header, {html: 'HYF Repositories', class: 'header-lable'});
+      const select = createAndAppend('select', header, { id: 'selectMenu' });
       if (err) {
         createAndAppend('div', root, { html: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
+        //createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
+        /* &&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&& */
+        data.sort((a, b) => a.name.localeCompare(b.name, 'fr', {ignorePunctuation: true}));
+        data.forEach((reopsitory, index) => {
+          createAndAppend('option', select, {html: reopsitory.name, value: index});
+        });
+        /* &&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&& */
+        const container = createAndAppend('div', root, { id: 'container'});
+        contentLeft(container, data, select);
+        contentRight(container, data, select);
       }
     });
   }
 
+
+
+
+
+  // function who creates a left content section
+  function contentLeft(container, data, select) {
+    const leftSection = createAndAppend('div', container, { id: 'leftSection', class: 'whiteframe' });
+    showTableRepo(leftSection, data, 0);
+
+    select.addEventListener('change', (event) => {
+      document.getElementById('leftSection').innerHTML = "";
+      const theEvent = event.target.value;
+      showTableRepo(leftSection, data, theEvent);
+    });
+  }
+
+  
+
+  // drow the table
+function showTableRepo(leftSection, data, dataIndex) {
+  // create table of the contents
+  const lTable = createAndAppend('table', leftSection);
+  // line one
+  const tr1 = createAndAppend('tr', lTable);
+  createAndAppend('td', tr1, {html: 'Repository :', class: 'lable'});
+  const td1 = createAndAppend('td', tr1);
+  let nameLink = '<a href="' + data[dataIndex].html_url + '" target="_blank">' + data[dataIndex].name + '</a>';
+  createAndAppend('p', td1, {html: nameLink });
+  // line two
+  const tr2 = createAndAppend('tr', lTable);
```

Comment:
Since you are repeating code here (create a `tr` with two `td`s as children, you could create a separate function to avoid repetition (the DRY principle: Don't Repeat Yourself), for instance:

```js
  function (table, label, value) {
    const tr = createAndAppend('tr', table);
    createAndAppend('td', tr, { html: `${label} :`, class: 'label' });
    createAndAppend('td', tr, { html: value });
  }
```